doc/ref/chap-type-method.md

```diff
@@ -27,6 +27,56 @@ This chapter in the [Oils Reference](index.html) describes YSH types and methods
 
 ### replace()
 
+Replace substrings or eggex matches within a string.
+
+    var mystr = "code is 1234"
+    = mystr => replace("code", "Code")  # => "Code is 1234"
+    = mystr => replace(/ d+ /, "<redacted>")  # => "code is <redacted>"
+
+Replacement expressions can use captures from the given eggex. Captured values
+can be read with `$1`, `$2`, etc. or, for named captures, with the names given
```

Comment:
I think it's more readable if you structure it like

... Refer to captured values by number ...

<example>

... Or by name ...

<example>

That makes the blocks of code shorter and easier to digest

---

Actually I would even do that with

Replace substring

<example>

or Eggex

<example>

i.e. so the examples are just 1 line